### PR TITLE
Remove some redundant variables and executions.

### PR DESCRIPTION
**Pull request by cchrewrite**
Note: If you encounter failed tests when building the project by running "mvn clean package", please remove "cool-core/src/test/java/com/nus/cool/core/refactor", and try to build the project again. The failed tests already exist before this pull request.

### CODE REVIEW

**Comment by cchrewrite** [comment]
Note: If you encounter failed tests when building the project by running "mvn clean package", please remove "cool-core/src/test/java/com/nus/cool/core/refactor", and try to build the project again. The failed tests already exist before this pull request.

File: cool-core/src/main/java/com/nus/cool/core/cohort/ExtendedCohortQuery.java [hunk 1]
```diff
@@ -80,13 +80,12 @@ public void setFillWithNextObserved(boolean fillWithNextObserved) {
...
 
     @JsonIgnore
-    public boolean isValid() throws IOException {
-        if ((birthSequence != null) &&
-            (birthSequence.isValid()) &&
-            (dataSource != null) &&
-            (ageField != null) &&
-            (measure != null)) return true;
-        else throw new IOException("[x] Invalid cohort query.");
+    public boolean isValid() {
```
Comment (cchrewrite) [inline cool-core/src/main/java/com/nus/cool/core/cohort/ExtendedCohortQuery.java:83]: Please revert this change, i.e., keep the original isValid() function with IOException.

**Comment by KimballCai** [comment]
We have refactored the COOL system codes that is why we have a new folder "cool-core/src/test/java/com/nus/cool/core/refactor". Later, we will delete all other codes with the old version (e.g., Extendedxxxxx.java).

There may be some unit tests that will fail due to the version conflict, and we will remove these codes with the old version and clean all the unit tests.

**Comment by KimballCai** [comment]
In the refactored codes, it is easier for developers to understand the codes, and we also have updated many functions.
e.g., support extending datasets, reducing redundancy storage, and so on.

File: cool-core/src/main/java/com/nus/cool/core/cohort/funnel/FunnelQuery.java [hunk 2]
```diff
@@ -21,40 +21,36 @@ public List<BirthSequence> getStages() {
...
+	// }
 
 	@JsonIgnore
-	public boolean isValid() throws IOException {
-		if ((stages != null) && (dataSource != null)){
-			return true;
-		} else{
-			throw new IOException("[x] Invalid funnel query.");
-		}
+	public boolean isValid() {
```
Comment (cchrewrite) [inline cool-core/src/main/java/com/nus/cool/core/cohort/funnel/FunnelQuery.java:53]: Please revert this change, i.e., keep the original isValid() function with IOException.

**Comment by cchrewrite** [comment]
KimballCai wrote:

>>> In the refactored codes, it is easier for developers to understand the codes, and we also have updated many functions. e.g., support extending datasets, reducing redundancy storage, and so on. <<<

Sure. That will be very helpful!

**Comment by cchrewrite** [comment]
Sure. That will be very helpful!

> KimballCai wrote: In the refactored codes, it is easier for developers to understand the codes, and we also have updated many functions. e.g., support extending datasets, reducing redundancy storage, and so on.

**Comment by KimballCai** [comment]
we will remove all the codes with the old version in the next two weeks.

**Comment by Zrealshadow** [comment]
Old-Version Code will be removed
